### ISSUE TITLE
Example of deepMock: TypeError: Cannot read property 'webRequest' of undefined

### ISSUE THREAD

**Issue by kabeleced77** [Lusito/mockzilla#10]
Hi, I really like to get mockzilla running for testing my web-extension. First of all I tried the given example [here](https://lusito.github.io/mockzilla/deep-mock.html#solution). I followed the instruction [here](https://lusito.github.io/mockzilla/setup.html#install-with-npm).

But I get 

    TypeError: Cannot read property 'webRequest' of undefined

       6 |
       7 | export function onBeforeRedirect(callback: () => void) {
    >  8 |   browser.webRequest.onBeforeRedirect.addListener(callback, { urls: [""] });
         |           ^
       9 |   return () => browser.webRequest.onBeforeRedirect.removeListener(callback);
      10 | }
      11 |

      at Object.onBeforeRedirect (src/my-web-extension.ts:8:11)
      at Object.<anonymous> (test/my-web-extension.spec.ts:37:30)

Could you please help? I don't know what to do else.

Thank you!

**Comment by Lusito** [comment]
Mockzilla itself does not mock browser by default. In the example you linked, the top-section does a `jest.mock` for webextension-polyfill-ts.

Mockzilla (this repository) does not by itself do anything for webextensions. If you have not seen it yet, [mockzilla-webextension](https://lusito.github.io/mockzilla-webextension/) is a fully-functioning solution for testing webextensions.

If you don't want mockzilla-webextension (for example if you are not using the webextension-polyfill), you can use mockzilla, but need to mock the browser instance yourself.

**Comment by kabeleced77** [comment]
Hi, thanks for answering (and understaning) my question.
I think I got it now. In fact I started with `mockzilla-webextension` as I am also using `webextension-polyfill-ts`. My "problem" was my own misunderstanding of mocking and I wanted to start with just `mockzilla`. 

Now I am using `mockzilla-webextension` based on your [example ](https://lusito.github.io/mockzilla-webextension/browser.html#solution). For my purpose I "only" needed a mock of `browser.storage.local`, e.g. `mockBrowser.storage.local.set.expect({ key: value }).andResolve()`.
Now I can also test my code using `mockzilla-webextension`.

Thanks for your work :).
